multiqc/utils/search_patterns.yaml

```diff
@@ -195,6 +195,12 @@ hicexplorer:
     num_lines: 5
 hicup:
     fn: 'HiCUP_summary_report*'
+hicpro:
+    - fn: '*.mmapstat'
```

Comment:
Normally I request that there is a separate search section for each _type_ of data, eg.:

```yaml
hicpro/mmapstat: fn: '*.mmapstat'
hicpro/mpairstat: fn: '*.mpairstat'
hicpro/mergestat: fn: '*.mergestat'
hicpro/mRSstat: fn: '*.mRSstat'
hicpro/assplit: fn: '*.assplit.stat'
```

This allows more fine-grained control of the searches, and ultimately better ability for users to configure the search patterns.

I just requested exactly this on another pull request, but here it is slightly less important as all files are parsed in an identical way and into a single data structure. However, I would prefer the above on principle (and we may be grateful for it in the future - search pattern identifiers should be as stable as possible).

It should be as simple as using the above yaml instead, then changing the parsing call to:

```python
for k in ['mmapstat','mpairstat','mergestat','mRSstat','assplit']:
    for f in self.find_log_files('hicpro/{}'.format(k)):
        self.parse_hicpro_stats(f)
```

You'll also need to stop creating a blank dict name for the sample in the parsing code each time, instead testing to see if the key exists already and creating a dict if not. The warning log message should then be triggered if a sample ID _and data key_ is already defined.

Does this make sense? Shout if you need help.

Reply:
ok.
By "stop creating a blank dict name for the sample in the parsing code each time", you mean in the  parse_hicpro_stats function ?

Replacing
`self.hicpro_data[s_name] = {}`
by
```
if s_name not in self.hicpro_data.keys():
    self.hicpro_data[s_name] = {}

```

Reply:
exactly 👍 

You'll need to change the overwriting warning too, instead warning if specific sample dict keys are overwritten.

Finally, you may need to think about how the `self.add_data_source()` is being called. Basically define `section` I think. See http://multiqc.info/docs/#printing-to-the-sources-file